src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -470,6 +475,27 @@ public void testBuildName() throws Exception {
   }
 
   @Test
+  public void testBuildWithPull() throws Exception {
+    final String dockerDirectory = Resources.getResource("dockerDirectory").getPath();
+    final String pullMsg = "Pulling from";
+
+    // Build once to make sure we have cached images.
+    sut.build(Paths.get(dockerDirectory));
+
+    // Build again with PULL set, and verify we pulled the base image
+    final AtomicBoolean pulled = new AtomicBoolean(false);
+    sut.build(Paths.get(dockerDirectory), "test", new ProgressHandler() {
+      @Override
+      public void progress(ProgressMessage message) throws DockerException {
+        if (message.stream().contains(pullMsg)) {
+          pulled.set(true);
+        }
+      }
+    }, PULL);
+    assertTrue(pulled.get());
```

Comment:
`docker build --pull` shows a "Pulling from" string in the stdout, but docker client doesn't seem to get it here.